lib/internal/url.js

```diff
@@ -767,7 +766,7 @@ ObjectDefineProperties(URLSearchParams.prototype, {
  * @returns {self is URL}
  */
 function isURL(self) {
-  return Boolean(self?.href && self.protocol && self.auth === undefined && self.path === undefined);
+  return typeof self === 'object' && self.href && self.protocol && self.auth === undefined && self.path === undefined;
```

Comment:
null is also an object, therefore passing null to this function break, won't it?

Reply:
addressed

Reply:
(I know it's what you meant but pedantically - null is not an object (its typeof is object, Google tried to fix it around ~2009 but it broke the internet))

Reply:
((I know it's what you meant but pedantically - it broke the web, not the internet :D ))